daemons/compute/server/servers/server_sim.go

```diff
@@ -95,3 +95,17 @@ func (s *simulatedServer) Delete(ctx context.Context, req *pb.DataMovementDelete
 		Status: pb.DataMovementDeleteResponse_DELETED,
 	}, nil
 }
+
+func (s *simulatedServer) List(ctx context.Context, req *pb.DataMovementListRequest) (*pb.DataMovementListResponse, error) {
+	if len(s.requests) > 0 {
+		// Get the uuids from the requests map
+		uids := make([]string, len(s.requests))
```

Comment:
Since you allocate the array up front can you avoid the `append` call with just an assignment `uids[i]=key.String()`?

Reply:
Is that preferable to create another variable (`i`) and increment it over using append? Should I not be allocating the array up front?

Reply:
ohh yeah, I forgot `range` on a map doesn't provide an index. OK - ignore my suggestion.

Reply:
One more thing though - if you are appending to an array of `len(s.requests)`, then the first `len(s.requests)` are marked empty in the array; so you're really returning an array 2x the expected value. I think since these are empty strings that they get filtered out by protobuf, but it should still be corrected.

I had to prove this out myself; here's an example https://go.dev/play/p/VDUcXQICPBo

I think what you could do is allocate an array of _lenght_ 0 but _capacity_ = len(s.requests). So it would be `make([]string, 0, len(s.requests))`